### ISSUE TITLE
tuf.LocalStore does not implement client.LocalStore

### ISSUE THREAD

**Issue by directionless** [theupdateframework/go-tuf#224]
Hi! I'm tinkering with `go-tuf` and I'm not sure I'm doing anything right yet, so forgive me if this is noise.

I tried doing something pretty simple:
```go
	localStore := tuf.FileSystemStore(*flLocalPath, fakePassphrases)

	transport := http.DefaultTransport.(*http.Transport).Clone()
	transport.TLSClientConfig = &tls.Config{}
	httpClient := &http.Client{Transport: transport}

	remoteStore, _ := client.HTTPRemoteStore(*flServer, nil, httpClient)

	tufClient := client.NewClient(localStore, remoteStore)
```

and this results in `tuf.LocalStore does not implement client.LocalStore (missing Close method)`

I think this is because #188 added Close, but did not add it to `tuf.FileSystemStore`. Though, I also notice that it's probably missing `DeleteMeta`

Is this an oversight? Is tuf.FileSystemStore not meant to be used for `client.LocalStore`? This generally seems kind of confusing.

**Comment by rdimitrov** [comment]
@directionless - The confusion comes from the fact that those two interfaces have the same name but come from different packages - one is for implementing the repository side of TUF and the other one is for the client. 

In this case `tuf.FIleSystemStore()` is a local storage implementation for the repository side whereas `client.NewClient()` creates a new client implementation expecting a different local storage (although it has the same name), i.e. `tuf_leveldbstore.FileLocalStore()`.

I think we should rename these so they are more explicit in their usage 👍

**Comment by directionless** [comment]
Ultimately I did figure that out, and I got some simple test code working. But yeah, the underlying names are confusing. And somewhat, I wonder why they're even different. Why not write a an implementation that handles both? 

(Anyhow, I'm going to close this. While I think the code is confusing, it's not really a big per se. Y'all can re-open if desired.)

**Comment by asraa** [comment]
> Why not write a an implementation that handles both?

I agree +1. The local stores for repositories and clients should be interoperable. It would also make testing easier :D
